File: elixir/darts/lib/darts.ex
defmodule Darts do
  @type position :: {number, number}

  @doc """
  Calculate the score of a single dart hitting a target
  """
  @spec score(position) :: integer
  def score({x, y}) do
    distance_squared = :math.sqrt(x * x + y * y)
    cond do
      distance_squared <= 1 -> 10
      distance_squared <= 5 -> 5
      distance_squared <= 10 -> 1
      true -> 0
    end
  end
end

File: java/darts/src/main/java/Darts.java
class Darts {
    int score(double xOfDart, double yOfDart) {
        double distanceSquared = xOfDart * xOfDart + yOfDart * yOfDart;
        if (distanceSquared <= 1) {
            return 10;
        } else if (distanceSquared <= 25) {
            return 5;
        } else if (distanceSquared <= 100) {
            return 1;
        } else {
            return 0;
        }
    }
}
